src/Taggable/TaggablePSR6PoolAdapter.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+namespace Cache\Taggable;
+
+use Cache\Taggable\TaggableItemInterface;
+use Cache\Taggable\TaggablePoolInterface;
+use Cache\Taggable\TaggablePoolTrait;
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+
+/**
+* 
+*/
+class TaggablePSR6PoolAdapter implements TaggablePoolInterface
+{
+    use TaggablePoolTrait;
+
+    private $cachePool;
+    private $tagStorePool;
+
+    private function __construct(CacheItemPoolInterface $cachePool, CacheItemPoolInterface $tagStorePool = null)
+    {
+        $this->cachePool = $cachePool;
+        if ($tagStorePool) {
+            $this->tagStorePool = $tagStorePool;
+        } else {
+            $this->tagStorePool = $cachePool;
+        }
+    }
+
+    public static function makeTaggable(CacheItemPoolInterface $cachePool, CacheItemPoolInterface $tagStorePool = null) // @TODO naming?
+    {
+        if ($cachePool instanceOf TaggablePoolInterface && $tagStorePool === null) {
+            return $cachePool;
+        }
+
+        return new self($cachePool, $tagStorePool);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getItem($key)
+    {
+        return TaggablePSR6ItemAdapter::makeTaggable($this->cachePool->getItem($key));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getItems(array $keys = array())
+    {
+        $items = $this->cachePool->getItems($keys);
+
+        $wrappedItems = [];
+        foreach ($items as $key => $item) {
+            $wrappedItems[$key] = TaggablePSR6ItemAdapter::makeTaggable($item);
+        }
+
+        return $wrappedItems;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function hasItem($key)
+    {
+        return $this->cachePool->hasItem($key);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function clear()
+    {
+        $ret = $this->cachePool->clear();
+        return $this->tagStorePool->clear() && $ret; // Is this acceptable?
```

Comment:
So the reason it wouldn't be okay is if you're passing in a pool that you're using for something else as your tag pool, that pool would be cleared when clearing this pool.

Ideally we'd be able to find all tags and do a multi-delete on those instead, leaving other data intact, or storing all of the tags in one single value instead, but both of those approaches have their own problems.